pkg/io/jsonio/unmarshal.go

```diff
@@ -91,43 +91,118 @@ func translateSegments(jsonSegments spec.SpecSegments, segmentType spec.Type) ma
 	cidrSegments := translateSegments(jsonSegments, spec.TypeCidr)
-	finalMap := make(map[string]map[string][]string)
+	finalMap := make(map[ir.ID]*ir.CidrSegmentDetails)
 	for segmentName, segment := range cidrSegments {
 		// each cidr saves the contained subnets
-		segmentMap := make(map[string][]string)
+		segmentMap := make(map[ir.CIDR]ir.CIDRDetails)
 		for _, cidr := range segment {
 			c, err := ipblock.FromCidr(cidr)
 			if err != nil {
 				return nil, err
 			}
-			validCidr, err := cidrContainedInVpc(*c, configDefs.AddressPrefixes)
+			vpcs, err := parseOverlappingVpcs(*c, configDefs.VPCs)
 			if err != nil {
 				return nil, err
 			}
-			if !validCidr {
-				return nil, fmt.Errorf("%s is not contained in the vpc", cidr)
-			}
 			subnets, err := configDefs.SubnetsContainedInCidr(*c)
 			if err != nil {
 				return nil, err
 			}
-			segmentMap[cidr] = subnets
+			segmentMap[ir.CidrFromString(cidr)] = ir.CIDRDetails{
+				OverlappingVPCs:  vpcs,
+				ContainedSubnets: subnets,
+			}
+		}
+		cidrSegmentDetails := ir.CidrSegmentDetails{
+			Cidrs: segmentMap,
 		}
-		finalMap[segmentName] = segmentMap
+		finalMap[segmentName] = &cidrSegmentDetails
 	}
 	return finalMap, nil
 }
 
-func translateIPMap(m map[string]string) map[string]ir.IP {
-	res := make(map[string]ir.IP)
+// read externals from conn-spec
+func translateExternals(m map[string]string) map[ir.ID]*ir.ExternalDetails {
+	res := make(map[ir.ID]*ir.ExternalDetails)
 	for k, v := range m {
-		res[k] = ir.IPFromString(v)
+		res[k] = &ir.ExternalDetails{IP: ir.IPFromString(v)}
 	}
 	return res
 }
 
+// replace all resources names in conn-spec to fully qualified name
+func replaceResourcesName(jsonSpec *spec.Spec, subnetSegments map[string][]ir.ID,
+	config *ir.ConfigDefs) (*spec.Spec, map[ir.ID]*ir.SubnetSegmentDetails, error) {
+	subnetsCache, ambiguousSubnets := inverseMapToFullyQualifiedName(config.Subnets)
+	nifsCache, ambiguousNifs := inverseMapToFullyQualifiedName(config.NIFs)
+	instancesCache, ambiguousInstances := inverseMapToFullyQualifiedName(config.Instances)
+	vpesCache, ambiguousVpes := inverseMapToFullyQualifiedName(config.VPEs)
+
+	// go over subnetSegments
+	finalSubnetSegments := make(map[ir.ID]*ir.SubnetSegmentDetails)
+	for segmentName, subnets := range subnetSegments {
+		subnetsNames := make([]ir.ID, 0)
+		VPCs := make([]ir.ID, 0)
+		for _, subnet := range subnets {
+			fullyQualifiedName, err := replaceResourceName(subnetsCache, ambiguousSubnets, subnet, spec.ResourceTypeSubnet)
+			if err != nil {
+				return nil, nil, err
+			}
+			subnetsNames = append(subnetsNames, fullyQualifiedName)
+			VPCs = append(VPCs, config.Subnets[fullyQualifiedName].VPC)
+		}
+		finalSubnetSegments[segmentName] = &ir.SubnetSegmentDetails{Subnets: subnetsNames, OverlappingVPCs: ir.UniqueIDValues(VPCs)}
+	}
+
+	var err error
+	// go over Spec
+	for i, conn := range jsonSpec.RequiredConnections {
```

Comment:
```suggestion
	for i := range jsonSpec.RequiredConnections {
		conn := &jsonSpec.RequiredConnections[i]
```
Avoids copying large struct, and can be used later to update the fully-qualified name:
```golang
    conn.Src.Name = fullyQualifiedSrc
```